### PR TITLE
Add reviewers to owners parsing

### PR DESCRIPTION
**Pull request by apelisse**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/2018)

### CODE REVIEW

File: mungegithub/mungers/approval-handler.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func (h *ApprovalHandler) hasApproval(filename string, approverSet sets.String,
 	paths := strings.Split(filename, "/")
 	p := ""
 	for i := 0; i < len(paths) && i < depth; i++ {
-		fileOwners := h.features.Repos.LeafAssignees(p)
+		fileOwners := h.features.Repos.Approvers(p)
```
Comment (foxish) [inline mungegithub/mungers/approval-handler.go]: p is always blank?

Comment (foxish) [inline mungegithub/mungers/approval-handler.go]: oh, nvm. Didn't see the rest of it :)

**Comment by foxish** [comment]
Looks good when quickly going over it. Will leave the detailed review to @grodrigues3

**Comment by grodrigues3** [comment]
LGTM

File: mungegithub/mungers/blunderbuss.go [hunk 2]
```diff
@@ -112,7 +112,7 @@ func (b *BlunderbussMunger) Munge(obj *github.MungeObject) {
 		// makes three buckets, we shouldn't have many 10k+
 		// line changes.
 		fileWeight = int64(math.Log10(float64(fileWeight))) + 1
-		fileOwners := b.features.Repos.LeafAssignees(*file.Filename)
+		fileOwners := b.features.Repos.LeafReviewers(*file.Filename)
```
Comment (apelisse) [inline mungegithub/mungers/blunderbuss.go]: I'm going to change this line and get this code merged. This is the only thing that is unsafe to merge.

**Comment by apelisse** [comment]
@foxish Made a change to the code so that it can be merged safely now.

Blunderbuss uses `Approvers`, which is merged with `Assignees` anyway in `repo-updates.go`. This is NOT changing the current behavior. The only change is that it reads reviewers, and doesn't use them.

Should be good to go.

**Comment by foxish** [comment]
The approval-handler is currently turned on for contrib. What's the expected change in behavior? Just that more people can approve? 
Also, what are your thoughts on moving the approval handler to prow, since `/lgtm` is over there now anyway?

**Comment by apelisse** [comment]
> The approval-handler is currently turned on for contrib. What's the expected change in behavior? Just that more people can approve? 

No, `blunderbuss` and `approval-handler` use the same list of assignees/approvers. Meaning that the same set of people can approve or lgtm. But also approve is not gated nor used at all. So technically: absolutely no change.

> Also, what are your thoughts on moving the approval handler to prow, since /lgtm is over there now anyway?

Absolutely, but it has some challenges (exposing who needs to approve what, moving the repo-update code there ..)

File: mungegithub/features/repo-updates.go [hunk 3]
```diff
@@ -184,15 +172,15 @@ func (o *RepoInfo) updateRepoUsers() error {
 	}
 	sha := out
 
-	o.assignees = map[string]sets.String{}
-	//o.owners = map[string]sets.String{}
+	o.approvers = map[string]sets.String{}
```
Comment (grodrigues3) [inline mungegithub/features/repo-updates.go]: you need to call make on the map?  and/or shouldn't it be sets.NewString()

**Comment by apelisse** [comment]
This is now integrated into `approval-process` pr/branch. Closing.

**Pull request closed without merging**